src/test/java/org/jenkinsci/plugins/github_branch_source/GitHubRepoSizeEstimatorTest.java

```diff
@@ -53,56 +76,23 @@ public void enableSystemCredentialsProvider() {
     }
 
     @Test
-    public void testExtensionApplicability() throws Exception {
-        GitHubRepoSizeEstimator.RepositorySizeGithubAPI repositorySizeGithubAPI = new GitHubRepoSizeEstimator.RepositorySizeGithubAPI();
-
-        String url = "https://github.com/jenkinsci/github-branch-source-plugin.git";
-
+    public void isApplicableToTest() throws Exception {
+        GitHubRepoSizeEstimator.RepositorySizeGithubAPI api =  j.jenkins.getExtensionList(GitToolChooser.RepositorySizeAPI.class).get(GitHubRepoSizeEstimator.RepositorySizeGithubAPI.class);
+        Item context = Mockito.mock(Item.class);
         store.addCredentials(Domain.global(), createCredential(CredentialsScope.GLOBAL, "github"));
         store.save();
 
-//        WorkflowJob p = j.jenkins.createProject(WorkflowJob.class, "p");
-//        p.setDefinition(new CpsFlowDefinition(
-//                "node {\n"
-//                        + "  checkout(\n"
-//                        + "    [$class: 'GitSCM', \n"
-//                        + "      userRemoteConfigs: [[credentialsId: 'github', url: $/" + url + "/$]]]\n"
-//                        + "  )"
-//                        + "}", true));
-//        WorkflowRun b = j.assertBuildStatusSuccess(p.scheduleBuild2(0));
-//        j.waitForMessage("using credential github", b);
-        List<UserRemoteConfig> repos = new ArrayList<>();
-        repos.add(new UserRemoteConfig(url, "origin", null, "github"));
-        FreeStyleProject projectWithMaster = setupProject(repos, Collections.singletonList(new BranchSpec("master")), null, false);
-        projectWithMaster.scheduleBuild2(0);
-
-        List<TopLevelItem> items =  j.jenkins.getItems();
+        githubApi.stubFor(
+                get(urlEqualTo("/"))
+                        .willReturn(
+                                aResponse()
+                                        .withHeader("Content-Type", "application/json; charset=utf-8")
+                                        .withBodyFile("../__files/body-(root)-XwEI7.json")));
 
-        assertThat(repositorySizeGithubAPI.isApplicableTo(url, items.get(0), "github"), is(true));
+        assertThat(api.isApplicableTo("https://github.com/rishabhBudhouliya/zoom-suspender.git", context, "github"), is(false));
```

Comment:
@bitwiseman The `GitHub.checkApiUrlValidity()` is reaching for `https://api.github.com/` and since the credentials are not valid, it returns a `bad credentials` error. 

To write the test cases, I would need to wiremock this and I have tried doing that looking at the test case examples provided by you but I have been unsuccessful in doing so.